VoodooPS2Trackpad/VoodooPS2ALPSGlidePoint.cpp

```diff
@@ -45,220 +177,231 @@ UInt32 ApplePS2ALPSGlidePoint::deviceType()
 UInt32 ApplePS2ALPSGlidePoint::interfaceID()
 { return NX_EVS_DEVICE_INTERFACE_BUS_ACE; };
 
-IOItemCount ApplePS2ALPSGlidePoint::buttonCount() { return 2; };
-IOFixed     ApplePS2ALPSGlidePoint::resolution()  { return _resolution; };
-bool IsItALPS(ALPSStatus_t *E6,ALPSStatus_t *E7);
+IOItemCount ApplePS2ALPSGlidePoint::buttonCount() { return _buttonCount; };
+IOFixed     ApplePS2ALPSGlidePoint::resolution()  { return _resolution << 16; };
+
+#define abs(x) ((x) < 0 ? -(x) : (x))
 
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 
-bool ApplePS2ALPSGlidePoint::init(OSDictionary * dict)
-{
+bool ApplePS2ALPSGlidePoint::init(OSDictionary *dict) {
+    
```

Comment:
I disabled trailing whitespace, but that did not change much.

![image](https://user-images.githubusercontent.com/28839925/153754561-9337fa1a-b6e1-4968-8841-b57743a8d9e2.png)

```diff
--- a/VoodooPS2Trackpad/VoodooPS2ALPSGlidePoint.cpp
+++ b/VoodooPS2Trackpad/VoodooPS2ALPSGlidePoint.cpp
@@ -655,19 +655,19 @@ void ApplePS2ALPSGlidePoint::alps_process_packet_v1_v2(UInt8 *packet) {
     
     // macOS does not support forward and back buttons
     /*
-    if (priv.flags & ALPS_FW_BK_1) {
-        back = packet[0] & 0x10;
-        forward = packet[2] & 4;
-    }
-    
-    if (priv.flags & ALPS_FW_BK_2) {
-        back = packet[3] & 4;
-        forward = packet[2] & 4;
-        if ((middle = forward && back)) {
-            forward = back = 0;
-        }
-    }
-    */
+     if (priv.flags & ALPS_FW_BK_1) {
+     back = packet[0] & 0x10;
+     forward = packet[2] & 4;
+     }
+     
+     if (priv.flags & ALPS_FW_BK_2) {
+     back = packet[3] & 4;
+     forward = packet[2] & 4;
+     if ((middle = forward && back)) {
+     forward = back = 0;
+     }
+     }
+     */
     
     ges = packet[2] & 1;
     fin = packet[2] & 2;
@@ -754,7 +754,7 @@ static void alps_get_bitmap_points(unsigned int map,
  * is greater than 0.
  */
 int ApplePS2ALPSGlidePoint::alps_process_bitmap(struct alps_data *priv,
-                              struct alps_fields *fields)
+                                                struct alps_fields *fields)
 {
     
     int i, fingers_x = 0, fingers_y = 0, fingers, closest;
@@ -1398,8 +1398,8 @@ unsigned char ApplePS2ALPSGlidePoint::alps_get_packet_id_v7(UInt8 *byte) {
 }
 
 void ApplePS2ALPSGlidePoint::alps_get_finger_coordinate_v7(struct input_mt_pos *mt,
-                                         UInt8 *pkt,
-                                         UInt8 pkt_id)
+                                                           UInt8 *pkt,
+                                                           UInt8 pkt_id)
 {
     mt[0].x = ((pkt[2] & 0x80) << 4);
     mt[0].x |= ((pkt[2] & 0x3F) << 5);
@@ -4177,9 +4177,9 @@ IOReturn ApplePS2ALPSGlidePoint::message(UInt32 type, IOService* provider, void*
             
             switch (pInfo->adbKeyCode)
             {
-                // don't store key time for modifier keys going down
-                // track modifiers for scrollzoom feature...
-                // (note: it turns out we didn't need to do this, but leaving this code in for now in case it is useful)
+                    // don't store key time for modifier keys going down
+                    // track modifiers for scrollzoom feature...
+                    // (note: it turns out we didn't need to do this, but leaving this code in for now in case it is useful)
                 case 0x38:  // left shift
                 case 0x3c:  // right shift
                 case 0x3b:  // left control
```